packages/cli/src/Server.ts

```diff
@@ -1693,197 +1708,200 @@ class App {
-						}
+				if (req.query.includeData !== 'true') {
+					const { data, id, ...rest } = credential;
 
-						const credentials = new Credentials(
-							result as INodeCredentialsDetails,
-							result.type,
-							result.nodesAccess,
-							result.data,
-						);
-						(result as ICredentialsDecryptedDb).data = credentials.getData(encryptionKey);
-					}
+					return {
+						id: id.toString(),
+						...rest,
+					};
+				}
 
-					(result as ICredentialsDecryptedResponse).id = result.id.toString();
+				const { data, id, ...rest } = credential;
 
-					return result as ICredentialsDecryptedResponse;
-				},
-			),
+				const encryptionKey = await UserSettings.getEncryptionKey();
+
+				if (!encryptionKey) {
+					throw new Error('No encryption key was found to decrypt the credentials!');
+				}
+
+				const coreCredential = new Credentials(
+					{ id: credential.id.toString(), name: credential.name },
+					credential.type,
+					credential.nodesAccess,
+					credential.data,
+				);
+
+				return {
+					id: id.toString(),
+					data: coreCredential.getData(encryptionKey),
+					...rest,
+				};
+			}),
 		);
 
 		// Returns all the saved credentials
 		this.app.get(
 			`/${this.restEndpoint}/credentials`,
 			ResponseHelper.send(
-				async (req: express.Request, res: express.Response): Promise<ICredentialsResponse[]> => {
-					const queryBuilder = Db.collections.Credentials!.createQueryBuilder('c');
-					queryBuilder.innerJoin('c.shared', 'shared');
-					queryBuilder.andWhere('shared.userId = :userId', {
-						userId: (req.user as User).id,
-					});
+				async (req: CredentialRequest.GetAll): Promise<ICredentialsResponse[]> => {
+					let credentials: ICredentialsDb[] = [];
 
-					if (req.query.filter) {
-						queryBuilder.andWhere(JSON.parse(req.query.filter as string));
-					}
+					const filter: Record<string, string> = req.query.filter
+						? JSON.parse(req.query.filter)
+						: {};
+
+					if (req.user.globalRole.name === 'owner') {
+						credentials = await Db.collections.Credentials!.find({
+							select: ['id', 'name', 'type', 'nodesAccess', 'createdAt', 'updatedAt'],
+							where: filter,
+						});
+					} else {
+						const shared = await Db.collections.SharedCredentials!.find({
+							relations: ['credentials'],
+							where: whereClause({
+								user: req.user,
+								entityType: 'credentials',
+							}),
+						});
 
-					queryBuilder.select([
-						'c.id',
-						'c.name',
-						'c.type',
-						'c.nodesAccess',
-						'c.createdAt',
-						'c.updatedAt',
-					]);
+						if (!shared.length) return [];
 
-					const results = (await queryBuilder.getMany()) as unknown as ICredentialsResponse[];
+						credentials = await Db.collections.Credentials!.find({
+							select: ['id', 'name', 'type', 'nodesAccess', 'createdAt', 'updatedAt'],
+							where: {
+								id: In(shared.map(({ credentials }) => credentials.id)),
+								...filter,
+							},
```

Comment:
aren't we getting all the credentials preloaded in the `shared` variable? can't we do the filtering/mapping without the db then?